src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AesAEAD.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography
+{
+    internal partial class AesAEAD
+    {
+        public static void CheckKeySize(int keySizeInBits)
+        {
+            if (keySizeInBits != 128 && keySizeInBits != 192 && keySizeInBits != 256)
+            {
+                throw new CryptographicException(SR.Cryptography_InvalidKeySize);
+            }
+        }
+
+        public static bool MatchesKeySizes(int size, KeySizes keySizes)
+        {
+            if (size < keySizes.MinSize || size > keySizes.MaxSize)
+                return false;
+
+            return ((size - keySizes.MinSize) % keySizes.SkipSize) == 0;
```

Comment:
This particular implementation doesn't work when a SkipSize of 0 is used.  But, that's semi-redundant against the previous comment.

Reply:
didn't expect that value since all values are hard-coded anyway - will unify with Helpers class which I haven't noticed before